central/scannerdefinitions/handler/updater_test.go

```diff
@@ -34,24 +34,74 @@ func assertOnFileExistence(t *testing.T, path string, shouldExist bool) {
 
 func TestUpdate(t *testing.T) {
 	filePath := filepath.Join(t.TempDir(), "dump.zip")
-	u := newUpdater(file.New(filePath), &http.Client{Timeout: 30 * time.Second}, defURL, 1*time.Hour)
-	// Should fetch first time.
-	require.NoError(t, u.doUpdate())
+
+	lastUpdatedTime := time.Now().UTC().Truncate(time.Second)
+	body := "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	responseLengths := [3]int{len(body), len(body), len(body)}
+	headContentLengths := [3]int{len(body), len(body), len(body)}
+	responseI := 0
+	responses := 1
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		responseI = min(responses-1, responseI)
+		w.Header().Set("Content-Length", fmt.Sprintf("%d", headContentLengths[responseI]))
+		w.Header().Set("Last-Modified", lastUpdatedTime.Format(http.TimeFormat))
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprint(w, body[:responseLengths[responseI]])
+		responseI++
+	}))
+
+	u := newUpdater(file.New(filePath), &http.Client{Timeout: 30 * time.Second}, server.URL, 1*time.Hour)
+	u.RetryDelay = 0
+
+	// Should fetch on first try.
+	assert.NotPanics(t, u.update)
```

Comment:
IIUC `NotPanics` only checks that the function can not panic: https://pkg.go.dev/github.com/stretchr/testify/assert#NotPanics, not that it will succeed on the first try.